### PR TITLE
Mobile Stories block (part 6): rewrote StoryEdit class as a function

### PR DESCRIPTION
**Pull request by mzorz**
Title has it - this is just a rewrite.

This PR builds on top of https://github.com/Automattic/jetpack/pull/17539, comes from the suggested comment by @jeherve in https://github.com/Automattic/jetpack/pull/17140#discussion_r505228667.

We're not adding new functionality in this PR, we just rewrote the class as a function. Decided to remove the code that otherwise previously existed in `componentWillUnMount` given the hook referenced there wasn't being really used (and with that, also removed the`doAction`, `hasAction` imports).

To test:
Test with the PRs related to the base branch, as follows (but updating the jetpack branch pointed to by the gutenberg-mobile submodule to this one)
Gutenberg-mobile wordpress-mobile/gutenberg-mobile#2747
WPANdroid wordpress-mobile/WordPress-Android#13174
1. run the host app
2. verify the blocks render correcly, upload and save progress is correctly represented, etc.

### CODE REVIEW

File: extensions/blocks/story/edit.native.js [hunk 1]
```diff
@@ -27,123 +26,77 @@ import { icon } from '.';
...
-			isSaveInProgress: false,
-			didUploadFail: false,
-			didSaveFail: false,
-		};
-	}
-
-	isUrlRemote( protocolForUrl ) {
+	function isUrlRemote( protocolForUrl ) {
 		return protocolForUrl === 'http:' || protocolForUrl === 'https:';
 	}
```
Comment (etoledom) [inline extensions/blocks/story/edit.native.js:49]: There is a `prependHTTP` function in `@wordpress/url` that might be used here instead.
Is `isUrlRemote` used at all? 🤔

**Review by etoledom — APPROVED** [review]
Looks and works great! 🎉 

Just a small comment that doesn't seem relevant for this particular PR.
✅

File: extensions/blocks/story/edit.native.js (hunk 1 above)
Comment (mzorz) [reply extensions/blocks/story/edit.native.js]: oops you're right! Removed it in d440e25fb4 👍

**Comment by mzorz** [comment]
Thanks once more for your review @etoledom ! 🎉 🙇 😄

**Pull request merged**